source/server/active_tcp_socket.cc

```diff
@@ -75,6 +74,33 @@ void ActiveTcpSocket::continueFilterChain(bool success) {
     bool no_error = true;
     if (iter_ == accept_filters_.end()) {
       iter_ = accept_filters_.begin();
+      if (listener_filter_max_read_bytes_ > 0) {
+        listener_filter_buffer_ = std::make_unique<Network::ListenerFilterBufferImpl>(
+            socket_->ioHandle(), listener_.dispatcher(),
+            [this]() {
+              socket_->ioHandle().close();
+              continueFilterChain(false);
+            },
+            [this]() {
+              // If the filter doesn't need any data, then return directly.
+              if ((*iter_)->maxReadBytes() == 0) {
+                return;
+              }
+              Network::FilterStatus status = (*iter_)->onData(*listener_filter_buffer_);
+              if (status == Network::FilterStatus::StopIteration) {
+                return;
+              }
+              continueFilterChain(true);
+            },
+            listener_filter_max_read_bytes_);
+        // when accept the connection, the socket already has the data, so trigger
```

Comment:
Sorry, probably I didn't describe it clearly. What I want to say is: We trigger the data peek at here since socket may already have data, we needn't wait for the next event loop.